bin/functional-tests/test_versioning.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
```

Comment:
IMHO this should be run through `black` to clean up the formatting. We could add black to pre-commit, which would make it easy for us all to forget about how to format our code in favor of having it automatically formatted.

```yaml
  - repo: https://github.com/psf/black
    rev: 20.8b1
    hooks:
      - id: black
```

Reply:
Some pathing situation I think I need to get black configured correctly to find at the path. I'm going to skip for now